activestorage/lib/active_storage/service/disk_service.rb

```diff
@@ -11,8 +11,10 @@ module ActiveStorage
   class Service::DiskService < Service
     attr_reader :root
 
-    def initialize(root:)
+    def initialize(root:, **options)
       @root = root
+
+      @public = options[:public]
```

Comment:
`public` is a legal identifier, so we can add it as a keyword arg without splatting:

```ruby
irb(main):003:0> class Foo
irb(main):004:1>   attr_reader :public
irb(main):005:1> 
irb(main):006:1>   def initialize(public: false)
irb(main):007:2>     @public = public
irb(main):008:2>   end
irb(main):009:1> end
=> :initialize
irb(main):010:0> Foo.new(public: true).public
=> true
```

Reply:
Thanks for the comments. TIL `public` is not a reserved keyword in Ruby.